ios/RNSScreen.mm

```diff
@@ -284,8 +397,75 @@ - (UIView *)reactSuperview
   return _reactSuperview;
 }
 
+- (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex
+{
+  [super insertReactSubview:subview atIndex:atIndex];
+}
+
+- (void)repositionFooter:(CGRect)footerFrame
+{
+  if (_footer == nil) {
+    return;
+  }
+
+  //  [_footer setFrame:CGRectMake(_footer.frame.origin.x,
+  //                               _footer.frame.origin.y,
+  //                               _footer.frame.size.width,
+  //                               _footer.frame.size.height)];
+  //
+}
+
+/// This is RNSScreenContentWrapperDelegate method, where we do get notified when React did update frame of our child.
+- (void)reactDidSetFrame:(CGRect)reactFrame forContentWrapper:(RNSScreenContentWrapper *)contentWrapepr
+{
+  if (self.stackPresentation != RNSScreenStackPresentationFormSheet || _didSetSheetAllowedDetentsOnController == YES) {
+    return;
+  }
+
+  _didSetSheetAllowedDetentsOnController = YES;
+
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_16_0) && \
+    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_16_0
+  if (@available(iOS 16.0, *)) {
+    // Sheet controller should be already in place, TODO: explain why
+    UISheetPresentationController *sheetController = _controller.sheetPresentationController;
+    if (sheetController == nil) {
+      RCTLogError(@"[RNScreens] sheetPresentationController is null when attempting to set allowed detents");
+      return;
+    }
+
+    if (_sheetAllowedDetents.count > 0 && _sheetAllowedDetents[0].intValue == -1) {
+      auto detents = [self detensFromMaxHeights:@[ [NSNumber numberWithFloat:reactFrame.size.height] ]];
+      [self setAllowedDetentsForSheet:sheetController to:detents animate:YES];
+    }
+  }
+#endif // Check for iOS >= 16
+}
+
 - (void)addSubview:(UIView *)view
 {
+  if ([view isKindOfClass:RNSScreenFooter.class]) {
+    _footer = (RNSScreenFooter *)view;
+    //    _footer.onLayout = ^(CGRect frame) {
+    //      [self updateFooterLayout];
+    //    };
+
+    //    [NSLayoutConstraint activateConstraints:@[
+    //      [NSLayoutConstraint constraintWithItem:_footer attribute:NSLayoutAttributeBottom
+    //      relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0],
+    //      [NSLayoutConstraint constraintWithItem:_footer attribute:NSLayoutAttributeLeft
+    //      relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeLeft multiplier:1.0 constant:0.0],
+    //      [NSLayoutConstraint constraintWithItem:_footer attribute:NSLayoutAttributeRight
+    //      relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeRight multiplier:1.0 constant:0.0],
+    //    ]];
+  }
+
+  if ([view isKindOfClass:RNSScreenContentWrapper.class] &&
+      self.stackPresentation == RNSScreenStackPresentationFormSheet) {
```

Comment:
The content wrapper will be always there from now on (as direct child of a Screen), however I do need a reference to it only in case we're in `formSheet` presentation.